cmake/HPX_AddModule.cmake

```diff
@@ -267,10 +267,33 @@ function(add_hpx_module libname modulename)
     # cmake-format: on
   endif()
 
-  target_link_libraries(
-    hpx_${modulename} PUBLIC ${${modulename}_MODULE_DEPENDENCIES}
-  )
-  target_link_libraries(hpx_${modulename} PUBLIC ${${modulename}_DEPENDENCIES})
+  if(${modulename}_MODULE_DEPENDENCIES)
+    # verify that all dependencies are from the same module category
+    foreach(dep ${${modulename}_MODULE_DEPENDENCIES})
+      # consider only module dependencies, not other targets
+      string(FIND ${dep} "hpx_" find_index)
+      string(FIND ${dep} "hpx_dependencies_" find_dep_index)
```

Comment:
This is a coincidence, but `hpx_dependencies_...` should in fact also not be in `MODULE_DEPENDENCIES`, would you mind changing the check to include that?

Reply:
I can do that. I will also change the then failing incantations.

Reply:
Thank you!